### ISSUE TITLE
Problem reading Shadows with EmbeddedId

### ISSUE THREAD

**Issue by tkuprevich** [javers/javers#886]
I'm using javers 5.7.2 and have a problem reading nested collection of `AgreementMember` objects in `Agreement`. It's a regilar `OneToMany` relation except `AgreementMember` has a composite primary key. Here are the classes:

`Agreement.java:`

```
@TypeName("Agreement")
public class Agreement
{
    @Id
    private UUID agreementId;

    private UUID locationId;

    @OneToMany(mappedBy = "agreement", cascade = CascadeType.ALL, orphanRemoval = true)
    @ShallowReference
    private List<AgreementMember> agreementMembers;

    //other fields ...
}
```

`AgreementMember.java`: with `AgreementMemberId` as primary key:

```
@TypeName("AgreementMember")
public class AgreementMember
{
    public AgreementMemberId getId()
    {
        return agreementMemberId;
    }


    @Embeddable
    @TypeName("AgreementMemberId")
    public static class AgreementMemberId implements Serializable
    {
        private UUID agreementId;
        private UUID memberId;
    }

    @EmbeddedId
    private AgreementMemberId agreementMemberId;

    //other fields ...
}
```
Here is a snapshot

```
[
   Snapshot   {
      commit:1.0,
      id:Agreement/a3ffda86-ba8b-4640-bd03-4ad945aa2546,
      version:1,
      (agreementId:a3ffda86-ba8b-4640-bd03-4ad945aa2546,
      agreementMembers:[
         AgreementMember/a3ffda86-ba8b-4640-bd03-4ad945aa2546,
         50527728-0f56-4403-8651-7f9eff2dda93
      ],
      locationId:1b1e5b47-ba10-4bb7-9f04-6b81d1c66422)
   }
]
```
And here is an exception when I'm trying to read shadows:

> org.javers.shadow.ShadowGraphBuilder#assembleShallowReferenceShadow java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $.agreementMemberId

And here is my test:

```
@Test
void testAgreement()
{
    //given
    Javers javers = JaversBuilder.javers()
            .withListCompareAlgorithm(ListCompareAlgorithm.SIMPLE)
            .build()

    Agreement agreement = new Agreement()
    agreement.agreementId = UUID.randomUUID()
    agreement.locationId = UUID.randomUUID()

    AgreementMember.AgreementMemberId agreementMemberId = new AgreementMember.AgreementMemberId()
    agreementMemberId.agreementId = agreement.agreementId
    agreementMemberId.memberId = UUID.randomUUID()

    AgreementMember agreementMember = new AgreementMember()
    agreementMember.agreementMemberId = agreementMemberId

    List<AgreementMember> agreementMemberList = new ArrayList<>()
    agreementMemberList.add(agreementMember)
    agreement.agreementMembers = agreementMemberList

    javers.commit("Agreement", agreement)

    JqlQuery query = QueryBuilder.byInstanceId(agreement.agreementId, Agreement.class).build()
    List<Shadow<Agreement>> shadows = javers.findShadows(query)
    println shadows
}
```

Any help is greatly appreciated. This might me misconfiguration or something, please advice.

**Comment by bartoszwalacik** [comment]
@tkuprevich you are cross-posting, and this makes our job harder.
https://stackoverflow.com/questions/58079203/javers-json-exception-when-reading-shadows-with-embedded-objects

**Comment by itisha** [comment]
I removed my post at stackoverflow and added failing case here
https://github.com/javers/javers/pull/889

Is that what I was supposed to do?

**Comment by itisha** [comment]
@bartoszwalacik should I reopen the case or do anything else? I'm still having the issue(

**Comment by bartoszwalacik** [comment]
ok, now I can look on this test case, stay tuned

**Comment by vsafronau** [comment]
any progress on this? We had the same issue.

**Comment by bartoszwalacik** [comment]
nope, this is an OSS project, if you want to speed up things, contribute a PR with the fic

**Comment by vsafronau** [comment]
thx

**Comment by tkuprevich** [comment]
This has been resolved by registering a custom toString() method
```
    @Data
    @Embeddable
    @TypeName( "AgreementMemberId" )
    public static class AgreementMemberId implements Serializable, JaversToString
    {
        @Column private UUID agreementId;

        @Column private UUID memberId;

        @Override
        public String javersToString()
        {
            return agreementId.toString() + ',' + memberId.toString();
        }
    }
```
and registering piece:
```
javers().registerValueWithCustomToString( JaversToString.class, JaversToString::javersToString )
```

**Comment by bartoszwalacik** [comment]
bug report confirmed

**Comment by bartoszwalacik** [comment]
as a quick workaround, you can remove `@ShallowReference`

**Comment by bartoszwalacik** [comment]
fixed in 5.8.3
